src/main/java/hudson/plugins/analysis/views/DetailFactory.java

```diff
@@ -263,5 +264,45 @@ private String createGenericTabUrl(final String link) {
     private int createHashCode(final String link, final String prefix) {
         return Integer.parseInt(StringUtils.substringAfter(link, prefix));
     }
-}
 
+    /**
+     * Creates a generic detail tab with the specified link.
+     *
+     * @param owner
+     *            the build as owner of the detail page
+     * @param annotations
+     *            the annotations to display
+     * @param url
+     *            the URL for the details view
+     * @param defaultEncoding
+     *            the default encoding to be used when reading and parsing files
+     * @return the detail view
+     * @deprecated use {@link #createTabDetail(Run, Collection, String, String)} instead
+     */
+    @Deprecated
+    protected TabDetail createTabDetail(final AbstractBuild<?, ?> owner, final Collection<FileAnnotation> annotations,
```

Comment:
You've gone to the trouble of maintaining strict binary compatibility for the two protected methods, but not for the public ones.